src/scripts/main.js

```diff
@@ -1 +1,54 @@
 'use strict';
+
+const firstPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+
+    setTimeout(() => {
+      reject(new Error(`First promise was rejected`));
+    }, 3000);
```

Comment:
The timeout for rejecting the first promise is set to 3000 milliseconds. This could lead to a race condition where the promise might be rejected before the 'mousedown' event occurs. Consider adding a flag to check if the promise has already been resolved before rejecting it.